builtin/aws/lambda/releaser.go

```diff
@@ -0,0 +1,323 @@
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "ResourceConflictException":
+				// function url config already exists. get it. maybe safe to continue
+				step.Update("Function url config already exists")
+
+				// retrieve existing function url to update state
+				if gfc, err := lambdasrv.GetFunctionUrlConfig(&lambda.GetFunctionUrlConfigInput{
+					FunctionName: aws.String(dep.FuncArn),
+				}); err != nil {
+					// this should not realistically occur
+					log.Error("Error getting function url config", "error", err)
+					return err
+				} else {
+					state.Url = *gfc.FunctionUrl
+					step.Update("Reusing existing Lambda URL: %q", state.Url)
+				}
+			default:
+				step.Update("Error creating function url config: %q, %q", aerr.Code(), aerr.Message())
+				return err
+			}
+		} else {
+			return err
+		}
+	} else {
+		// update state
+		state.Url = *cfo.FunctionUrl
+		step.Update("Created Lambda URL: %q", state.Url)
+	}
+
+	step.Done()
+
+	return nil
+}
+
+func (r *Releaser) Release(
+	ctx context.Context,
+	log hclog.Logger,
+	src *component.Source,
+	ui terminal.UI,
+	dep *Deployment,
+) (*Release, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	// Create our resource manager and create
+	rm := r.resourceManager(log)
+	if err := rm.CreateAll(
+		ctx, log, sg, ui, src,
+		dep,
+	); err != nil {
+		log.Info("Error creating resources", "error", err)
+		return nil, err
+	}
+
+	// Get our function url state to verify
+	fnUrlState := rm.Resource("function_url").State().(*Resource_FunctionUrl)
+	log.Info("Function URL state", "url", fnUrlState.Url)
+	if fnUrlState == nil {
+		return nil, status.Errorf(codes.Internal, "function url state is nil, this should never happen")
+	}
+
+	return &Release{
+		Url:           fnUrlState.Url,
+		FuncArn:       dep.FuncArn,
+		VerArn:        dep.VerArn,
+		ResourceState: rm.State(),
+	}, nil
+}
+
+func (r *Releaser) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	sess, err := utils.GetSession(&utils.SessionConfig{
+		Logger: log,
+	})
+	if err != nil {
+		return err
+	}
+
+	rm := r.resourceManager(log)
+
+	// Destroy All
+	return rm.DestroyAll(ctx, log, sg, ui, sess)
+}
+
+// ReleaserConfig is the configuration structure for the Releaser.
+type ReleaserConfig struct {
+	// "AWS_IAM" or "NONE"
+	AuthType string `hcl:"auth_type,optional"`
+	// todo(kevinwang): CORS
+}
+
+func (r *Releaser) Status(
+	ctx context.Context,
+	log hclog.Logger,
+	release *Release,
+	ui terminal.UI,
+) (*sdk.StatusReport, error) {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+
+	s := sg.Add("Gathering health report for lambda url deployment: %q", release.Url)
+	defer s.Done()
+
+	report := sdk.StatusReport{
+		External: true,
+	}
+
+	// check the function url status
+	log.Info("Checking function url status...", "url", release.Url)
+	if resp, err := http.Get(release.Url); err != nil {
```

Comment:
Is there an AWS SDK function here we could use instead to validate the creation/existence of a lamba url instead of http.Get?

Reply:
I don't _think_ there's a _healthcheck-like_ API for Lambda 🤔. (Makes sense the its ephemeral nature)
- [lambda - available commands](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/lambda/index.html#available-commands)
- (looking for something like [elbv2::describe-target-health](https://docs.aws.amazon.com/cli/latest/reference/elbv2/describe-target-health.html))

1. I'm now realizing that `AWS_IAM` auth type would cause this unsigned http.Get to be return a `503`
Maybe 
2. Maybe `wait` or `invoke` are viable options...

Reply:
Does the `lambda` `get-function` help at all? I don't know if it has any kind of health or status in the return response. Otherwise yeah maybe the way to go is an `http.Get`. Might be another good question to ask the team this week and see what they say.

Reply:
`get-function` doesn't return any useful status information. I'll shoot a message to the Waypoint team slack for some opinions!